documentation/src/main/asciidoc/ch09.asciidoc

```diff
@@ -370,6 +370,100 @@ include::{sourcedir}/org/hibernate/validator/referenceguide/chapter09/Bootstrapp
 Refer to <<section-fail-fast>> and <<section-programmatic-api>> to learn more about the fail fast
 mode and the constraint declaration API.
 
+[[section-script-evaluator-factory]]
+==== Configuring script evaluator factory (`ScriptEvaluatorFactory`)
+
+For constrains like `@ScriptAssert` and `@ParameterScriptAssert` it might be useful to configure
+how scripts are evaluated, and which scripting engines are used. This can be done by setting
+a custom implementation of `ScriptEvaluatorFactory`. In particular this is important for
+OSGi environments (or similar to it) where user might face issues with dynamic class loading
+and https://www.jcp.org/en/jsr/detail?id=223[JSR-223]. It also would allow to use any custom
+script evaluation, not necessarily based on the https://www.jcp.org/en/jsr/detail?id=223[JSR-223].
+
+===== Programmatic configuration
+There are two ways to change the default `ScriptEvaluatorFactory` programmatically.
+First one is to use `hibernate.validator.script_evaluator_factory` property as described
```

Comment:
I don't think I would advertise the property configuration in the programmatic API. I would only use it for the XML configuration.